compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -360,7 +384,10 @@ object JavaParsers {
         in.token match {
           case AT if (in.lookaheadToken != INTERFACE) =>
             in.nextToken()
-            annotation()
+            annotation() match {
+              case Some(anno) => annots :+= anno
```

Comment:
Since you're appending to a List, you could use a `ListBuffer`

Reply:
Done